### PR TITLE
Beginning of new class on functions

### PR DESCRIPTION
**Pull request by thesteve0**

### CODE REVIEW

File: basic-postgresql-devel/basicfunctions/02-name-parameters.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+'''
+select brilliance(rank => 1);
+'''{{execute}}
+
+
+## Wrap Up
+
+**NOTE** No two functions can have the same name UNLESS they have  different parameter signatures. 
+For example, you can't have two functions named _myfunction_ unless one is myfunction() 
+and the other is myfunction(myparam varchar). What this also means is that you can overload a function to
```
Comment (davecramer) [inline basic-postgresql-devel/basicfunctions/02-name-parameters.md]: somewhat misleading... you can actually have two functions with parameters but with different types.

File: basic-postgresql-devel/basicfunctions/01-first-function.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Our First Function
+
+Let's lay a little groundwork so we can write our first function. There are some basic pieces all functions, regardless 
+of programming language, need in order to work.
+
+## Basic Pieces of Every Function
+
+Just like functions in other languages, PostgreSQL functions has certain structure and syntax. Let's avoid dealing
```
Comment (davecramer) [inline basic-postgresql-devel/basicfunctions/01-first-function.md]: s/has/have

File: basic-postgresql-devel/basicfunctions/03-return-types.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+PostgreSQL allows you to specify the "direction" data flows into a function based on how you declare the parameter. By 
+default all parameters are specified as IN parameters, meaning data is only passed into the function with the parameter. 
+
+If you declare a parameter as OUT then data will be returned. You can have more than one OUT parameter in a function. Using a 
+RETURNS <type> AS statement when you have OUT parameters is optional. If you 
+use the RETURNS <type> AS statement with your function, it's type must match the type of our OUT parameters. 
+
+An INOUT parameter means it will be used to pass data in and then return data through this parameter name as well
+
+For our function let's rid of the RETURNS <type> AS statement and add another OUT parameter
```
Comment (davecramer) [inline basic-postgresql-devel/basicfunctions/03-return-types.md]: get rid of

**Comment by thesteve0** [comment]
Keep em coming I will get them in the morning (hopefully before the class).
I will add a third example to the overloading to show you can have a(), a(x
int), and a(x varchar)

**Comment by thesteve0** [comment]
Will be fixed in this latest commit

**Pull request merged**